CDCgov/prime-reportstream
# #5920 Cleanup FHIR empty properties generated by HL7 translation

This PR updates the HL7 to FHIR configuration to ignore various properties when they're empty.

Test Steps:
1. Before switching branches, make a copy of the test file at `/src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_20220518-0001.fhir`
2. Check out the branch for this PR.
3. Run `./gradlew package` and `testIntegration` to verify the build and tests pass.
4. Compare the `sample_co_1_20220518.fhir` that you backed up with the one that now exists on the PR. Search by `{ }` (note the space) on both files.
5. Confirm that you see a lot of them on the backed up file, but none on the new file.
6. Other than that, any other differences must be UUID and timestamp related.

## Changes
- Added skipping logic on various FHIR resources to reduce the amount of junk data.
- Updated test files to remain accurate to real conversion logic.

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
